homeassistant/components/asuswrt/sensor.py

```diff
@@ -203,6 +203,6 @@ def native_value(self) -> float | int | str | None:
         """Return current state."""
         descr = self.entity_description
         state: float | int | str | None = self.coordinator.data.get(descr.key)
-        if state is not None and descr.factor and isinstance(state, (float, int)):
+        if state is not None and descr.factor and isinstance(state, float | int):
```

Comment:
Side note: The union syntax is 100 - 50 % slower on Python 3.10 and Python 3.11 than the tuple syntax, according to the issue for this feature in ruff.

Reply:
For reference, this is the issue mentionned above: https://github.com/charliermarsh/ruff/issues/2923
See also [PEP 604](https://peps.python.org/pep-0604/#isinstance-and-issubclass)

Reply:
> Side note: The union syntax is 100 - 50 % slower on Python 3.10 and Python 3.11 than the tuple syntax, according to the issue for this feature in ruff.

Does it make sense then to replace it if it's slower? The way I understand PEP 604, it's a new option to use the Union syntax. However the old way won't go anywhere anytime soon.

Reply:
We can add `UP038` to ignore list, and I can revert commit 65a884ad9a01727a275cc72a1d82812da6719b0d if that is what we want.

Reply:
I'm all in favor of modern syntaxes, but we don't have to pay the performance price right now, which affects 95% of the userbase. I think we can reconsider once we run Python 3.11 on our main distribution.

For now, I rather have this one on the ignore list.

Reply:
Done